driver-kafka/src/main/java/io/openmessaging/benchmark/driver/kafka/KafkaBenchmarkDriver.java

```diff
@@ -88,47 +87,102 @@ public String getTopicNamePrefix() {
     }
 
     @Override
-    public CompletableFuture<Void> createTopic(String topic, int partitions) {
+    public CompletableFuture<Void> createTopic(String topic, int partitions, boolean onlyValidate) {
         return CompletableFuture.runAsync(() -> {
             try {
-                admin.createTopics(Arrays.asList(new NewTopic(topic, partitions, config.replicationFactor))).all()
-                        .get();
+                ListTopicsResult allTopics = admin.listTopics();
+                Set<String> topicsStrs = allTopics.names().get();
+                if (topicsStrs.contains(topic)) {
+                    if (!onlyValidate) {
+                        DeleteTopicsResult deleteResult = admin.deleteTopics(Collections.singleton(topic));
+                        deleteResult.all().get();
+                        Thread.sleep(TimeUnit.SECONDS.toMillis(3)); // TODO: Grrrr...more polling
+                        createNewKafkaTopic(topic, partitions);
+                    }
+                } else {
+                    createNewKafkaTopic(topic, partitions);
+                }
+
+
+                // Make sure topic gets a leader
```

Comment:
What is the goal for waiting until we get a leader? If the leader is not ready, the producer will get blocked for a little while, right? In any case the measurements from the "warm-up" period are not being collected.

Reply:
If you generate enough topics in kafka, you'll see leader-election take longer than the warm-up period.

Reply:
Good point. Let me try to do this by using a "barrier". 
 * For Pulsar consumer the barrier is triggered immediately after the consumer is created (which is when it's ready to go)
 * For Kafka, we can wait for the partition assignment

Reply:
@afalko I gave a shot in https://github.com/streamlio/messaging-benchmark/commit/02e40c4aa3c412ffc0a3edd551033dbdab6a5ab2

In addition of having a leader for the topic, I had the problem of having the consumer to "ready" to receive messages. That is important when probing the max-sustainable throughput (with `producerRate : 0`), since the logic gets broken if the consumer doesn't start receiving messages immediately.

Reply:
Thanks, I'll merge that into my branch!